x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtRealm.java

```diff
@@ -149,640 +159,455 @@ public JwtRealm(
-                        + "].",
-                    e
-                );
-                urlException = e;
-            }
-        } else {
-            urlException = new Exception(
-                "Parse URL not attempted for ["
-                    + jwkSetPath
-                    + "] for setting ["
-                    + RealmSettings.getFullSettingKey(realmConfig, JwtRealmSettings.JWKSET_PATH)
-                    + "]. Only HTTPS URL or local file are supported."
-            );
-        }
-        // If local file parsing succeeds then return right away, otherwise save the exception and fall through to next step.
-        final Exception pathException;
-        try {
-            // Use separate lines of code, in case any are null. That allows NPE stack trace line number to show exactly which one is null.
-            final Environment environment = realmConfig.env();
-            final Path directoryPath = environment.configFile();
-            final Path filePath = directoryPath.resolve(jwkSetPath);
-            // Check file exists, accessible, and non-empty. TODO Maybe return parsed contents here instead of later in a watcher thread.
-            final String fileContents = Files.readString(filePath, StandardCharsets.UTF_8);
-            if (Strings.hasText(fileContents) == false) {
-                throw new Exception(
-                    "Empty file ["
-                        + jwkSetPath
-                        + "] for setting ["
-                        + RealmSettings.getFullSettingKey(realmConfig, JwtRealmSettings.JWKSET_PATH)
-                        + "]."
-                );
-            }
-            return new Tuple<>(null, filePath); // RETURN URL AS NULL AND PATH AS NON-NULL (i.e. URL Exception is discarded)
-        } catch (Exception e) {
-            pathException = new Exception(
-                "Error loading local file ["
-                    + jwkSetPath
-                    + "] for setting ["
-                    + RealmSettings.getFullSettingKey(realmConfig, JwtRealmSettings.JWKSET_PATH)
-                    + "].",
-                e
-            );
+    public void ensureInitialized() {
+        if (this.delegatedAuthorizationSupport == null) {
+            throw new IllegalStateException("Realm has not been initialized");
         }
-        // Throw SettingsException with the two suppressed exceptions.
-        final SettingsException settingsException = new SettingsException(
-            "Invalid value [" + jwkSetPath + "] for setting " + RealmSettings.getFullSettingKey(realmConfig, JwtRealmSettings.JWKSET_PATH)
-        );
-        settingsException.addSuppressed(urlException);
-        settingsException.addSuppressed(pathException);
-        throw settingsException;
     }
 
     @Override
     public void initialize(final Iterable<Realm> allRealms, final XPackLicenseState xpackLicenseState) {
+        LOGGER.trace("Initializing realm [" + super.name() + "]");
```

Comment:
Another comment on the logging discussion. I don't think that a trace log that was helpful to diagnose a test failure during development is a good candidate for a log line necessarily. Or that we should be adding too much logging in case something ends up being helpful at some point in some case in production. If we genuinely think that a log is going to be helpful in production then it should not be TRACE, it should be DEBUG. 

Of course more logging is helpful, but we need to strike a balance between no logs and logging after "We are here, the execution state is X and the object contents are Y" every other line. I think this steers a little bit more to the right of that axis than we usually do. On the other hand, we have no agreed standards ( apart from consistency with existing code ) on how we log, so I have nothing concrete to make my case upon :)

Reply:
Removed trace log. Not worth effort to debate it so I removed it.